x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportCloseJobAction.java

```diff
@@ -331,10 +338,13 @@ public void onResponse(PersistentTasksCustomMetaData.PersistentTask<?> task) {
 
                             @Override
                             public void onFailure(Exception e) {
-                                final int slot = counter.incrementAndGet();
-                                failures.set(slot - 1, e);
-                                if (slot == numberOfJobs) {
-                                    sendResponseOrFailure(request.getJobId(), listener, failures);
+                                if ((e instanceof ResourceNotFoundException &&
+                                    Strings.isAllOrWildcard(new String[]{request.getJobId()})) == false) {
+                                    final int slot = counter.incrementAndGet();
+                                    failures.set(slot - 1, e);
+                                    if (slot == numberOfJobs) {
```

Comment:
I think this `if` statement needs to be after the scope of `if ((e instanceof ResourceNotFoundException &&
	                                    Strings.isAllOrWildcard(new String[]{request.getJobId()})) == false) {` ends, otherwise we won't respond to the caller if the race occurs on the very last job.